MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -5734,19 +5734,19 @@ public int findAtBPartsAvailabilityLevel(IAcquisitionWork acquisition, StringBui
          * If the unit is not assigned to a contract, use the least restrictive active
          * contract. Don't restrict parts availability by contract if it has not started.
          */
-        if (hasActiveContract() && (contract == null)) {
+        if (hasActiveContract()) {
             for (Contract c : getActiveContracts()) {
-                if (c instanceof AtBContract && 
-                        c.getStartDate().isBefore(currentDay) && 
-                        (contract == null || 
+                if ((c instanceof AtBContract) && 
+                        (c.getStartDate().isBefore(currentDay) || c.getStartDate().equals(currentDay)) && 
+                        ((contract == null) || 
                         (((AtBContract) c).getPartsAvailabilityLevel() > contract.getPartsAvailabilityLevel()))) {
                     contract = (AtBContract) c;
                 }
             }
         }
         
         // if we have a contract and it has started
-        if ((null != contract) && contract.getStartDate().isBefore(currentDay)) {
+        if ((null != contract) && (contract.getStartDate().isBefore(currentDay) || contract.getStartDate().equals(currentDay))) {
```

Comment:
The getStartDate stuff is not required because of the use of getActiveContracts. Additionally, currentDay should never be directly used outside of the increase day.

Reply:
I'm not sure about that getStartDate thing, because the contract may not come from the "activeContracts" loop.

Reply:
Missed that part, you are correct. Still the latter currentDay bit holds, plus you can just do a !contract.getStartDate().isBefore(getLocalDate()) to test if it is either before or equal to the current day

Reply:
Yeah, that's a lot cleaner.